qgispluginci/release.py

```diff
@@ -94,76 +102,40 @@ def create_archive(
                     count=parameters.changelog_number_of_entries
                 )
                 if content:
-                    replace_in_file(
-                        f"{parameters.plugin_path}/metadata.txt",
-                        r"^changelog=.*$",
-                        f"changelog={content}",
-                    )
+                    metadata.set("general", "changelog", content)
             except Exception as exc:
                 # Do not fail the release process if something is wrong when parsing the changelog
-                replace_in_file(
-                    f"{parameters.plugin_path}/metadata.txt",
-                    r"^changelog=.*$",
-                    "",
-                )
+                metadata.remove_option("general", "changelog")
                 logger.warning(
                     f"An exception occurred while parsing the changelog file: {exc}",
                     exc_info=exc,
                 )
     else:
         # Remove the changelog line
-        replace_in_file(f"{parameters.plugin_path}/metadata.txt", r"^changelog=.*$", "")
+        metadata.remove_option("general", "changelog")
 
     # set version in metadata
-    replace_in_file(
-        f"{parameters.plugin_path}/metadata.txt",
-        r"^version=.*$",
-        f"version={release_version}",
-    )
-
+    metadata.set("general", "version", release_version)
     # Commit number
-    replace_in_file(
-        f"{parameters.plugin_path}/metadata.txt",
-        r"^commitNumber=.*$",
-        f"commitNumber={len(list(repo.iter_commits()))}",
-    )
-
+    metadata.set("general", "commitNumber", str(sum(1 for _ in repo.iter_commits())))
```

Comment:
`commitNumber` `commitSha1` and `dateTime` are not necessarily in the `general` section because they are not "official" items from a `metadata.txt` according to QGIS documentation https://docs.qgis.org/3.40/en/docs/pyqgis_developer_cookbook/plugins/plugins.html#metadata-txt
It's up to the PyQGIS developer.

https://opengisch.github.io/qgis-plugin-ci/usage/cli_release.html#additional-metadata